### DISCUSSION TITLE
JsonElement / JsonProperty - get utf-8 property name without allocations

### DISCUSSION THREAD

**Discussion by exyi** [dotnet/runtime#99839] [General]
Hello, I'm looking for a way to enumerate a `JsonElement` object properties, ideally free of heap allocations. Currently, I see two options for accessing the property name - `JsonProperty.Name` and `JsonProperty.NameEquals` (if I know the possible names). The `Name` property however allocates a new string every time, and it doesn't seem to do any interning.

```C#
foreach (var p in obj.EnumerateObject())
{
    p.Name // returns newly allocated string
    p.NameEquals("abc"u8) // I don't know all possible names
}
```

Is there any other way? I'm looking for something like `Utf8JsonReader.CopyString(Span<byte> utf8Destination)`. I actually want the name in utf-8, so decoding it using the `Name` property just to encode it back feels quite dumb :]

**Comment by huoyaoyuan** [comment]
There's a similar discussion for `Value` in #77666. Escaping may make things complex.

**Reply by exyi** [reply]
Thanks, I didn't notice this one. I suppose the potential `CopyString(Span<byte>)` should also be added to JsonProperty to allow allocation free reading of properties. The Utf8JsonReader.CopyString unescapes the string, so I suppose escaping should not be issue here. The #30207 issue is actually closer to what I need, but is unfortunately closed as "won't do" :/
